Toys and stuff

JStheguy opened the pull request:
-Adds another foam dart gun based on the double barrled shotgun.
-Cap Gun, Foam Dart Crossbow, Foam Dart Shotgun have their own sprites.
-Foam Dart crossbow now only has a dart capacity of 1.
-Foam dart guns now come in boxes with extra darts.
-Adds Spare Foam Dart box.
-Adds 36 action figures sprited by MrSnapwalk. (Arcade prizes)
-And the foam dart/foam dart gun box things are now arcade prizes in
place of the plain foam dart guns.

JStheguy committed:
Toys and stuff

-Adds another foam dart gun based on the double barrled shotgun.
-Cap Gun, Foam Dart Crossbow, Foam Dart Shotgun have their own sprites.
-Foam Dart crossbow now only has a dart capacity of 1.
-Foam dart guns now come in boxes with extra darts.
-Adds Spare Foam Dart box.
-Adds 36 action figures sprited by MrSnapwalk. (Arcade prizes)
-And the foam dart/foam dart gun box things are now arcade prizes in
place of the plain foam dart guns.

Iamgoofball commented on code/game/objects/items/toys.dm:
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
+	icon_state = "nuketoy"
+	var/cooldown = 0
+	var/toysay = "What the fuck did you do?"
+	var/obj/item/toy/accessory/hastoy = null
```
what's going on with this?

Neerti replied:
it's an abstract, for inheritance.  It's not meant to be seen by players normally.

Iamgoofball commented on code/game/objects/items/toys.dm:
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
 
+/obj/item/toy/figure/attack_self(mob/user as mob)
+	if(!cooldown)
+		user << "<span class='notice'>The [src] says \"[toysay]\".</span>"
```
it should say this to everyone in the area.

You should also be able to rename the toys for shits and giggles.

Atskadan replied:
no

Iamgoofball replied:
Why not?

Atskadan replied:
no

Iamgoofball replied:
yes

Commit:
Update arcade.dm

Okay now should work

JStheguy commented:
Okay it should work now.

RemieRichards commented:
no... these are not toy figures of crew/syndies...

WOW. want.

RemieRichards commented on code/game/objects/items/toys.dm:
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
+	name = "Chaplain action figure"
+	desc = "A \"Space Life\" brand Chaplain action figure."
+	icon_state = "chaplain"
+	toysay = "Praise Space Jesus!"
```
could actually have this check against the current round's deity, and if there isn't one, set it to "Space Jesus" as default.

MrSnapwalk commented:
Here, have a picture of all the action figures. These were fun to make!
![](http://puu.sh/9yulZ/e090e9808f.png)

RemieRichards commented:
@MrSnapwalk  Hnnnnng OMG they are SOOOO CUTE

Commit:
CHAPLAIN ACTION FIGURES UPDATE

Due to complaints by religious groups Nanotrasen sanctioned "Space Life" brand Chaplain action figures no longer make reference to any specific deity or religion.

JStheguy commented:
Read: I'm too fucking lazy at this point to try and figure out how fucking chaplain deity shit works nor try to figure out how to make that shit change what the action figure says.

MrSnapwalk commented on code/game/objects/items/toys.dm:
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
+	name = "Captain action figure"
+	desc = "A \"Space Life\" brand Captian action figure."
+	icon_state = "captain"
+	toysay = "Any Heads of Staff?"
```
Should be "Any heads of staff?".

MrSnapwalk commented on code/game/objects/items/toys.dm:
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
+	name = "Chef action figure"
+	desc = "A \"Space Life\" brand Chef action figure."
+	icon_state = "chef"
+	toysay = "Bartender is a tastey burger."
```
You misspelled tasty in this line.

Atskadan replied:
This line doesn't make sense either

MrSnapwalk commented on code/game/objects/items/toys.dm:
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
+	name = "Warden action figure"
+	desc = "A \"Space Life\" brand Warden action figure."
+	icon_state = "warden"
+	toysay = "17 Minutes for coughing at an officer!"
```
"Minutes" should be lower case.

Atskadan replied:
Numbers which start a sentence are always spelled out, it should be "Seventeen"

MrSnapwalk commented:
Pointed out some spelling and capitalization mistakes, it's up to you if you feel like fixing them. Either way, good job! Looking forward to having these in-game.

Atskadan commented on code/game/objects/items/toys.dm:
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
+	name = "Cyborg action figure"
+	desc = "A \"Space Life\" brand Cyborg action figure."
+	icon_state = "borg"
+	toysay = "I live again."
```
all caps pls
and periods after every word

Atskadan commented on code/game/objects/items/toys.dm:
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
+	name = "Chaplain action figure"
+	desc = "A \"Space Life\" brand Chaplain action figure."
+	icon_state = "chaplain"
+	toysay = "Join my religion, you fucks."
```
something else please like
"Glory to X" X being the deity

RemieRichards replied:
I actually removed the global deity variable when I cleaned up Bible vars sitting on the damn Game ticker.

I'd suggest

```
var/deity = pick(bibledeitynames) //bibledeitynames IS the name of the random deity list.
toysay = "Glory to [deity]!"
```

Atskadan commented on code/game/objects/items/toys.dm:
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
+	name = "Ian action figure"
+	desc = "A \"Space Life\" brand Ian action figure."
+	icon_state = "ian"
+	toysay = "Aruuu!"
```
ian isn't a wolf
he says AUUUU or ARF
arf would be better

Atskadan commented on code/game/objects/items/toys.dm:
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
+	name = "Detective action figure"
+	desc = "A \"Space Life\" brand Detective action figure."
+	icon_state = "detective"
+	toysay = "The airlock has grey jumpsuit and insulated glove fibers on it."
```
THIS airlock

Atskadan commented on code/game/objects/items/toys.dm:
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
+	name = "Chief Engineer action figure"
+	desc = "A \"Space Life\" brand Chief Engineer action figure."
+	icon_state = "ce"
+	toysay = "Oh god the singularity is loose!"
```
why not give this to the engineer and change it to "Wire the solars!"

Atskadan commented on code/game/objects/items/toys.dm:
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
+	name = "Quartermaster action figure"
+	desc = "A \"Space Life\" brand Quartermaster action figure."
+	icon_state = "qm"
+	toysay = "Please sign this form in triplicate and we will see about geting you a welding mask within 3 buisness days."
```
misspelled business

Atskadan commented on code/game/objects/items/toys.dm:
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
+	name = "Lawyer action figure"
+	desc = "A \"Space Life\" brand Lawyer action figure."
+	icon_state = "lawyer"
+	toysay = "My client is not guilty!"
```
no no make it OBJECTION

Atskadan commented on code/game/objects/items/toys.dm:
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
+	toysay = "Big stompy mechs!"
+
+/obj/item/toy/figure/scientist
+	name = "Roboticist action figure"
```
should be scientist

Commit:
ENGILSH 010 WITH JSTHEGUY

WHY AM I SO BAD AT ENGLISH I'VE SPOKEN AND WROTE IN IT MY ENTIRE LIFE

ChangelingRain commented on code/game/objects/items/toys.dm:
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
+	name = "Chef action figure"
+	desc = "A \"Space Life\" brand Chef action figure."
+	icon_state = "chef"
+	toysay = "Bartender is a tasty burger."
```
You, uh, do mean Pun-Pun right?

Atskadan commented:
pls update

JStheguy commented:
Why update?

Atskadan commented:
uh.. so that i'll merge it?

JStheguy commented:
Srs what is there to update.

Atskadan commented:
there's still an error in one of your descriptions and i would like (but not require) that you make the changes i suggested as well

RemieRichards commented:
you saw nothing.

Atskadan commented:
remie it can be merged

RemieRichards commented:
Oh right.

Atskadan commented:
i saw absolutely nothing.

Commit:
Update toys.dm

JStheguy commented:
Yeah is good.

Commit:
Update toys.dm

Commit:
OH SUE ME I MISSPELLED A THING

JStheguy commented:
OKAY YEAH IS GOOD TO GO NOW(?)

Atskadan commented:
no you still have the thing with the chaplain quote
remie even told you how to do it

Commit:
ABSOLUTELY NEEDED CHAPLAIN FIGURE DEITY THING ADD

JStheguy commented:
FISSION MAILED

Commit:
EMIT CONTINENT

Commit:
WORK YOU BASTARD

JStheguy commented:
Yeah I cant get this shit to work.

Commit:
WELP THAT DIDN'T WORK

Atskadan commented:
are you getting any compile errors or anything?

JStheguy commented:
Yes.
code/game/objects/items/toys.dm:1026:error: toydeity: invalid variable
code/game/objects/items/toys.dm:1026:error: : expected a constant expression

(Toydeity being just the deity thing from remies original snippit of code)
(and line 1026 being the chaplains Toysay line.)

(Right now the current code on my branch no longer has the deity whatthefuck)

Iamgoofball commented:
a var defined like that can't be random on definiton

it needs to be defined and THEN made random on the new:

SO:

/obj/dick/butt
     desc = "hm, goku is a"
     var/dickinbutt = ""

/obj/dick/butt/New()
     ..()
     dickinbutt = "[pick(bibledeitynames)]"
     desc = "hm, goku is a [dickinbutt]"

is what works, as a variable can't be randomized on definition like you had it before.

Commit:
THANKS BASED GOOFBALL

Commit:
Fukkin fuhgadabout that

JStheguy commented:
Okay is good now?
If so you can merge.
